scheduler/filter/utils.go

```diff
@@ -0,0 +1,89 @@
+package filter
+
+import (
+	"regexp"
+	"strings"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+type comparison int
+
+const (
+	equ = comparison(iota)
+	neg
+	gte
+	lte
+)
+
+func parse(k, v string) (string, string, comparison, bool) {
+	// default comparison mode
+	mode := equ
+
+	// support case of constraint:k==v
+	// try to check = on both sides, "k= : = : v" and "k : = : =v", to make sure it works everywhere
```

Comment:
Using "=", it's possible for an entry "k==v" to be split as:
1. "k=" as key and "v" as value
2. "k" as key and "=v" as value
I just make sure to cover these cases.